### ISSUE TITLE
Unable to select player Class on new game start

### ISSUE THREAD

**Issue by Daniel-Leontiev** [Sumwunn/DoomRPG#62]

**Comment by FoxMcloud5655** [comment]
First of all, are your mods loaded in the right order?

**Comment by Daniel-Leontiev** [comment]
I would hope so, Is there a reference to what the correct order should be?
currently using:

DoomRL_Arsenal_1.1.2
DoomRL_Monsters_Beta_7.2

In that order, tried the other way too. with different wads. win no luck.

Using this version: gzdoom-x64-g4.3pre-108-gc107657ea

**Comment by Daniel-Leontiev** [comment]
This is my bat file:

`gzdoom.exe -skill 3 -warp 1 +sv_cheats 1 +logfile "Doom RPG.log" -file "DRLA_1.1.2.pk3" "DRLM_7.2.pk3" DoomRPG\DoomRPG DoomRPG\DoomRPG-RLArsenal DoomRPG\DoomRPG-RLMonsters DoomRPG\DoomRPG-Brightmaps DoomRPG\DoomRPG-Extras`

All works well, but can't select player class on new game.

**Comment by ALLEN2CE** [comment]
having the same issue. while using 1.1.1, the log file contains tons of script errors and i get this console message when the first map loads: GiveInventory: Unknown item type DRPGDRLAActive. I get the same message while using 1.1.2 and the game is basically unplayable with errors all over the place. I never had this before while using previous versions of DRLA

**Comment by ALLEN2CE** [comment]
does using 1.1.2 make a difference compared to 1.1.1?

**Comment by Daniel-Leontiev** [comment]
No there is no difference.

**Comment by FoxMcloud5655** [comment]
My order that works is as follows:
* DoomRPG
* DoomRL Arsenal
* DoomRL Arsenal - Monsters
* DoomRPG - Brightmaps
* DoomRPG - RLArsenal
* DoomRPG - RLMonsters

RLArsenal is version 1.1.1 and DoomRPG was built as of 2019-07-06.

**Comment by ALLEN2CE** [comment]
Im assuming that you don't use the DRPG launcher because that does not work, at least not for me. Maybe the fix is to load DRPG before the rla + monsters pk3/wad and have the rest load afterwards.

**Comment by FoxMcloud5655** [comment]
Exactly why the above is my load order :)

**Comment by Daniel-Leontiev** [comment]
Thank you! It's all good now. This is my bat:

`
gzdoom.exe -skill 3 -warp 1 +sv_cheats 1 +logfile "Doom RPG.log" -file DoomRPG\DoomRPG "DRLA_1.1.2.pk3" "DRLM_7.2.pk3" DoomRPG\DoomRPG-Brightmaps DoomRPG\DoomRPG-Extras DoomRPG\DoomRPG-RLArsenal DoomRPG\DoomRPG-RLMonsters
`

**Comment by Forevener** [comment]
> Im assuming that you don't use the DRPG launcher because that does not work, at least not for me. Maybe the fix is to load DRPG before the rla + monsters pk3/wad and have the rest load afterwards.

Hello! I've worked on the launcher pretty hard and made sure that it works fine for me. Could you explain the issues you having, please? I would like to fix them.

**Comment by ALLEN2CE** [comment]
> > Im assuming that you don't use the DRPG launcher because that does not work, at least not for me. Maybe the fix is to load DRPG before the rla + monsters pk3/wad and have the rest load afterwards.
> 
> Hello! I've worked on the launcher pretty hard and made sure that it works fine for me. Could you explain the issues you having, please? I would like to fix them.

For the launcher, it runs DRPG and RLA in the wrong order. At least for me, for some reason it loads everything in this order.

DoomRL Arsenal
DoomRL Arsenal - Monsters
DoomRPG
DoomRPG - Brightmaps
DoomRPG - RLArsenal
DoomRPG - RLMonsters

For some reason, i'm not quite sure why it does that. Is there a way to make it load DRPG first before wads?

**Comment by Forevener** [comment]
> > > Im assuming that you don't use the DRPG launcher because that does not work, at least not for me. Maybe the fix is to load DRPG before the rla + monsters pk3/wad and have the rest load afterwards.
> > 
> > 
> > Hello! I've worked on the launcher pretty hard and made sure that it works fine for me. Could you explain the issues you having, please? I would like to fix them.
> 
> For the launcher, it runs DRPG and RLA in the wrong order. At least for me, for some reason it loads everything in this order.
> 
> DoomRL Arsenal
> DoomRL Arsenal - Monsters
> DoomRPG
> DoomRPG - Brightmaps
> DoomRPG - RLArsenal
> DoomRPG - RLMonsters
> 
> For some reason, i'm not quite sure why it does that. Is there a way to make it load DRPG first before wads?

Well, this behaviour is based on the recommendations from the wiki: https://github.com/Sumwunn/DoomRPG/wiki#running - where WADs are coming first, and DRPG+patches are coming last (because they should override everything else). Which problems are you having with that load order, and which load order do you actually use?

**Comment by ALLEN2CE** [comment]
> > > > Im assuming that you don't use the DRPG launcher because that does not work, at least not for me. Maybe the fix is to load DRPG before the rla + monsters pk3/wad and have the rest load afterwards.
> > > 
> > > 
> > > Hello! I've worked on the launcher pretty hard and made sure that it works fine for me. Could you explain the issues you having, please? I would like to fix them.
> > 
> > 
> > For the launcher, it runs DRPG and RLA in the wrong order. At least for me, for some reason it loads everything in this order.
> > DoomRL Arsenal
> > DoomRL Arsenal - Monsters
> > DoomRPG
> > DoomRPG - Brightmaps
> > DoomRPG - RLArsenal
> > DoomRPG - RLMonsters
> > For some reason, i'm not quite sure why it does that. Is there a way to make it load DRPG first before wads?
> 
> Well, this behaviour is based on the recommendations from the wiki: https://github.com/Sumwunn/DoomRPG/wiki#running - where WADs are coming first, and DRPG+patches are coming last (because they should override everything else). Which problems are you having with that load order, and which load order do you actually use?

The problem is that according to the load order of the https://github.com/Sumwunn/DoomRPG/wiki#running page, loading the WADs first is a problem. OP's issue was with the same load order as the wiki (-file "DRLA_1.1.2.pk3" "DRLM_7.2.pk3" **DoomRPG\DoomRPG DoomRPG\DoomRPG-RLArsenal DoomRPG\DoomRPG-RLMonsters DoomRPG\DoomRPG-Brightmaps DoomRPG\DoomRPG-Extras**) instead of the correct way which is (-file **DoomRPG\DoomRPG** "DRLA_1.1.2.pk3" "DRLM_7.2.pk3" DoomRPG\DoomRPG-Brightmaps DoomRPG\DoomRPG-Extras DoomRPG\DoomRPG-RLArsenal DoomRPG\DoomRPG-RLMonsters)

Basically, DRPG should be loaded first then the RLA then the patches. For some reason, OP's load order would have worked in the past versions of the mod from what i remembered a year or two ago.

**Comment by Daniel-Leontiev** [comment]
This is the correct load order:
`gzdoom.exe +sv_cheats 1 +logfile "Doom RPG.log" -file "DRLA\DRLA_1.1.2.pk3" "DRLA\DRLM_7.2.pk3" DoomRPG\DoomRPG DoomRPG\DoomRPG-Brightmaps DoomRPG\DoomRPG-Extras DoomRPG\DoomRPG-RLArsenal DoomRPG\DoomRPG-RLMonsters`
